lupa/__init__.py

```diff
@@ -1,36 +1,34 @@
 from __future__ import absolute_import
 
+from contextlib import contextmanager
 
-# We need to enable global symbol visibility for lupa in order to
-# support binary module loading in Lua.  If we can enable it here, we
-# do it temporarily.
+# Find the implementation with the latest Lua version available.
+_newest_lib = None
 
-def _try_import_with_global_library_symbols():
+
+@contextmanager
+def eager_global_linking():
```

Comment:
"eager" linking isn't the intention here. Yes, we use `RTLD_NOW`, but only to enable `RTLD_GLOBAL`. What we really want to enable here is Lua module loading. I propose
```suggestion
def allow_lua_module_loading():
```
so that you can write
```python
import lupa
with lupa.allow_lua_module_loading():
    from lupa import lua54
```
Admittedly, that looks a bit funny, but I don't really see how we could do better if the import needs to be wrapped.